kernel/mouse/src/lib.rs

```diff
@@ -41,9 +81,13 @@ pub fn init(mut mouse: PS2Mouse<'static>, mouse_queue_producer: Queue<Event>) ->
         "PS/2 mouse IRQ was already in use! Sharing IRQs is currently unsupported."
     })?;
 
+    // Initialize the mouse packet bytes, which will be filled by 3-4 interrupts,
+    // depending on the MouseId
+    let packet_bytes = PacketBytes::new(mouse.packet_size());
```

Comment:
here's where you'd specify the const generic parameter, either 3 or 4 depending upon mouse ID.

Reply:
Nice idea, didn't consider const generics.